src/main/java/org/mule/extension/http/internal/request/HttpRequester.java

```diff
@@ -181,16 +214,21 @@ private void doRequestWithRetry(HttpExtensionClient client, HttpRequesterConfig
                 scheduler.submit(() -> consumePayload(result));
                 doRequest(client, config, uri, method, streamingMode, sendBodyMode, followRedirects,
                           authentication, responseTimeout, responseValidator, transformationService,
-                          requestBuilder, false, muleContext, scheduler, notificationEmitter,
-                          streamingHelper, callback, injectedHeaders, correlationId);
+                          requestCreator, false, muleContext, scheduler, notificationEmitter,
+                          streamingHelper, callback, injectedHeaders);
               }, () -> {
                 responseValidator.validate((Result) result, httpRequest, streamingHelper);
 
                 Result<Object, HttpResponseAttributes> freshResult = httpResponseToResult
                     .convert(config, muleContext, response, entity, resultInputStreamSupplier, httpRequest.getUri());
 
-                profilingDataProducer
-                    .ifPresent(profilingDataProducer -> profilingDataProducer.triggerProfilingEvent(result, correlationId));
+                profilingDataProducer.ifPresent(
+                                                profilingDataProducer -> profilingDataProducer
+                                                    .triggerProfilingEvent(result,
```

Comment:
i think this would be more readable if you extract `correlationId` to a variable